### ISSUE TITLE
Investigate AbortController() pre-emptively stopping requests

### ISSUE THREAD

**Issue by MichaelSwigerAtBentley** [iTwin/admin-components-react#128]
In [Desktop View Test in iTwin/viewer](https://github.com/iTwin/viewer/tree/master/packages/apps/desktop-viewer-test) when attempting to get imodels for a given iTwin, the api call is stalling and being aborted:

![Image](https://github.com/user-attachments/assets/ee0b2a3e-7986-4771-a045-bd77fdb98d78)

This causes the iModels page to never load, leaving 3 loading animations cycling forever, as described in this [Issue](https://github.com/iTwin/viewer/issues/336)

Removing the 
`signal: abortController.signal`
from [useIModelData.ts#L127](https://github.com/iTwin/admin-components-react/blob/dca65649b402d4d023d00d2dbc6b4457b7731f5e/packages/modules/imodel-browser/src/containers/iModelGrid/useIModelData.ts#L127) corrects the issue.

We should investigate why this is interfering with the request. 

Additional info:
Running the request with the same accessToken and headers does return the iModels as normal, ruling out a clientId or auth issue.

**Comment by MichaelSwigerAtBentley** [comment]
@aruniverse Nam and I were able to reproduce the user's issue, and it seems related to the abortController as mentioned above.

**Comment by aruniverse** [comment]
is this an issue with react 18 strict mode?

**Comment by MichaelSwigerAtBentley** [comment]
> is this an issue with react 18 strict mode?

Yes, that did it. Removing the strict mode corrects the issue. Should I remove it from the desktop-test-app?

**Comment by aruniverse** [comment]
> > is this an issue with react 18 strict mode?
> 
> Yes, that did it. Removing the strict mode corrects the issue. Should I remove it from the desktop-test-app?

Remove it from the template and suggest that as a fix to the user

**Comment by MichaelSwigerAtBentley** [comment]
Closing this issue with PR: https://github.com/iTwin/viewer/pull/339

**Comment by aruniverse** [comment]
Re-opening, as there still is an issue with this custom hook in react 18 strict mode
